src/shogun/statistical_testing/internals/Block.cpp

```diff
@@ -47,32 +46,33 @@ Block::Block(CFeatures* feats, index_t index, index_t size) : m_feats(feats)
 	SG_REF(m_feats);
 
 	// create a shallow copy and subset current block separately
-	CFeatures* block=FeaturesUtil::create_shallow_copy(feats);
-	ASSERT(block->ref_count()==0);
+	CFeatures* block=feats->shallow_subset_copy();
```

Comment:
For our present requirements, this method suffices. But it does not clone the entire subset and attach it to the shallow copy, rather just pushes the last working index into it. So, if someone wants to pop out the subsets in the copy, it can't be done. Should the need arise, will have to add that.

Also, maybe this method could use a better name. Maybe `shallow_copy_with_active_subset()` or so?